Source/missiles.cpp

```diff
@@ -1375,12 +1375,19 @@ void AddLightningWall(Missile &missile, const AddMissileParameter &parameter)
 	UpdateMissileVelocity(missile, parameter.dst, 16);
 	missile._miAnimFrame = GenerateRnd(8) + 1;
 	missile._mirange = 255 * (missile._mispllvl + 1);
-	if (missile._misource < 0) {
+	switch (missile.sourceType()) {
+	case MissileSource::Trap:
 		missile.var1 = missile.position.start.x;
 		missile.var2 = missile.position.start.y;
-	} else {
-		missile.var1 = Players[missile._misource].position.tile.x;
-		missile.var2 = Players[missile._misource].position.tile.y;
+		break;
+	case MissileSource::Player: {
+		Player &player = *missile.sourcePlayer();
+		missile.var1 = player.position.tile.x;
+		missile.var2 = player.position.tile.y;
+	} break;
+	case MissileSource::Monster:
+		assert(missile.sourceType() != MissileSource::Monster);

```

Comment:
I liked the one with false, if we got here, then we don't really have to check this :D

Reply:
The problem is that you get no context in the error message:
![image](https://user-images.githubusercontent.com/204594/180663588-9767f5f3-4162-4bbc-a131-cfa7d8ebca3a.png)

If we hit this code we are already in a place we shouldn't be